Test/DependencyCollector/FunctionalTests/FuncTest/HttpTests.cs

```diff
@@ -153,212 +155,245 @@ public void MyTestCleanup()
         public void TestRddForAzureSdkBlobAspx451()
         {
-            if (!RegistryCheck.IsNet451Installed)
-            {
-                Assert.Inconclusive(".Net Framework 4.5.1 is not installed");
-            }
+            EnsureNet451Installed();
 
             this.ExecuteAzureSDKTests(DeploymentAndValidationTools.Aspx451TestWebApplication, 1, "blob", "http://127.0.0.1:11000");           
         }
 
         [TestMethod]
-        [DeploymentItem("..\\TestApps\\ASPX451\\App\\", DeploymentAndValidationTools.Aspx451AppFolder)]
+        [TestCategory(TestCategory.Net451)]
+        [DeploymentItem(Aspx451TestAppFolder, DeploymentAndValidationTools.Aspx451AppFolder)]
         public void TestRddForAzureSdkQueueAspx451()
         {
-            if (!RegistryCheck.IsNet451Installed)
-            {
-                Assert.Inconclusive(".Net Framework 4.5.1 is not installed");
-            }
+            EnsureNet451Installed();
 
             this.ExecuteAzureSDKTests(DeploymentAndValidationTools.Aspx451TestWebApplication, 1, "queue", "http://127.0.0.1:11001");           
         }
 
         [TestMethod]
-        [DeploymentItem("..\\TestApps\\ASPX451\\App\\", DeploymentAndValidationTools.Aspx451AppFolder)]
+        [TestCategory(TestCategory.Net451)]
+        [DeploymentItem(Aspx451TestAppFolder, DeploymentAndValidationTools.Aspx451AppFolder)]
         public void TestRddForAzureSdkTableAspx451()
         {
-            if (!RegistryCheck.IsNet451Installed)
-            {
-                Assert.Inconclusive(".Net Framework 4.5.1 is not installed");
-            }
+            EnsureNet451Installed();
 
             this.ExecuteAzureSDKTests(DeploymentAndValidationTools.Aspx451TestWebApplication, 1, "table", "http://127.0.0.1:11002");           
         }
 
         [TestMethod]
-        [DeploymentItem("..\\TestApps\\ASPX451\\App\\", DeploymentAndValidationTools.Aspx451AppFolderWin32)]
+        [TestCategory(TestCategory.Net451)]
+        [DeploymentItem(Aspx451TestAppFolder, DeploymentAndValidationTools.Aspx451AppFolderWin32)]
         public void TestRddForWin32ApplicationPool()
         {
-            if (!RegistryCheck.IsNet451Installed)
-            {
-                Assert.Inconclusive(".Net Framework 4.5.1 is not installed");
-            }
+            EnsureNet451Installed();
+
             this.ExecuteSyncHttpTests(DeploymentAndValidationTools.Aspx451TestWebApplicationWin32, true, 1, AccessTimeMaxHttpInitial);
         }
 
         #endregion 451
-                          
+
+        #region Core
+
+        private static void EnsureDotNetCoreInstalled()
+        {
+            string error = "";
+            DotNetCoreProcess process = new DotNetCoreProcess("--version")
+                .RedirectStandardErrorTo((string errorMessage) => error += errorMessage)
+                .Run();
+
+            if (process.ExitCode.Value != 0 || !string.IsNullOrEmpty(error))
```

Comment:
Would it be good to also check for a minimum version? (e.g. 1.0.1 or 1.1.0 etc.)